src/ui/atoms/button/Button.tsx

```diff
@@ -42,22 +54,34 @@ export const Button: React.FC<TButtonProps> = ({
   backgroundColor = 'primary',
   disabled = false,
   label,
+  leftIcon,
+  rightIcon,
+  icon,
   onClick
 }: TButtonProps): JSX.Element => (
   <button
     className={classNames('okp4-button-main', {
       [`variant-${variant}`]: true,
       [`size-${size}`]: true,
       [`background-color-${backgroundColor}`]: true,
-      disabled: !!disabled
+      disabled,
+      icon: variant === 'icon'
     })}
     disabled={disabled}
     onClick={onClick}
     title={label}
     type="button"
   >
-    <Typography as="div" color="highlighted-text" fontSize="small" fontWeight="light">
-      {label}
-    </Typography>
+    {variant === 'icon' ? (
+      icon
+    ) : (
+      <div className="okp4-button-label">
```

Comment:
The className `okp4-button-label` should carry more context in its naming as it does not only refers to the label

Reply:
You are write, I've changed it too `okp4-button-description`. It's hard to come up with a more precise name as the content rendering depends on if the property values exists or not.